### DISCUSSION TITLE
Will multiple compactors with vertical compaction break Thanos data?

### DISCUSSION THREAD

**Discussion by ethernoy** [thanos-io/thanos#4870] [Questions & Answers]
Hi, I am implementing a blue-green deployment of a set of services in our cluster, including Thanos. In this case, there may be at most 2 compactor instances running and connecting to the same object storage. I already have vertical compaction enabled over Prometheus external replica label and receive label. I wonder will this set-up break Thanos data?

**Comment by michael-burt** [comment]
You can have multiple Compactors operating on the same bucket, but they must be operating on unique block streams, or set of external labels. You cannot have two compactors that are operating on the same set of external labels, this will break eventually, and potentially corrupt your data.
